src/queries/__tests__/role.test.tsx

```diff
@@ -135,3 +142,470 @@ describe('supports name option', () => {
     );
   });
 });
+
+describe('supports accessibility states', () => {
+  describe('disabled', () => {
+    test('returns a disabled element when required', () => {
+      const { getByRole } = render(
+        <TouchableOpacity
+          accessibilityRole="button"
+          accessibilityState={{ disabled: true }}
+        />
+      );
+
+      expect(getByRole('button', { disabled: true })).toBeTruthy();
+    });
+
+    test('returns the correct element when only one matches all the requirements', () => {
+      const { getByRole } = render(
+        <>
+          <TouchableOpacity
+            testID="correct"
+            accessibilityRole="button"
+            accessibilityState={{ disabled: true }}
+          >
+            <Text>Save</Text>
+          </TouchableOpacity>
+          <TouchableOpacity testID="wrong" accessibilityRole="button">
+            <Text>Save</Text>
+          </TouchableOpacity>
+        </>
+      );
+
+      expect(
+        getByRole('button', { name: 'Save', disabled: true }).props.testID
+      ).toBe('correct');
+    });
+
+    test('returns an implicitly enabled element', () => {
+      const { getByRole } = render(
+        <TouchableOpacity accessibilityRole="button"></TouchableOpacity>
+      );
+
+      expect(getByRole('button', { disabled: false })).toBeTruthy();
+    });
+
+    test('returns an explicitly enabled element', () => {
+      const { getByRole } = render(
+        <TouchableOpacity
+          accessibilityRole="button"
+          accessibilityState={{ disabled: false }}
+        ></TouchableOpacity>
+      );
+
+      expect(getByRole('button', { disabled: false })).toBeTruthy();
+    });
+
+    test('does not return disabled elements when querying for non disabled', () => {
+      const { queryByRole } = render(
+        <TouchableOpacity
+          accessibilityRole="button"
+          accessibilityState={{ disabled: true }}
+        ></TouchableOpacity>
+      );
+
+      expect(queryByRole('button', { disabled: false })).toBe(null);
+    });
+
+    test('returns elements using the built-in disabled prop', () => {
```

Comment:
Can we simplify this test to only contain a single `TouchableOpacity` with `disabled={true}`?

Reply:
Those three RN component use different implementation behind the hood, so my goal was to test all of them (`TouchableOpacity` and `TouchableHighlight` extend `TouchableWithoutFeedback`), and I felt like covering our bases.

On the other hand I do see your point, since we're just testing RN internal implementation.

I'll keep the three ones, if you feel strongly about that (or splitting it in three tests) let me know).

Reply:
It makes sense to test all three `TouchablesXxx` and I would also metion `Pressable` in that context. Maybe you can use `test.each` to automate testing over these component variants?

Reply:
I tried, but the difference in types (`Button` doesn't use `children`), it leads to pretty ugly code, and I'd say that it's not worth it, it'll make the tests harder to read, and here it's fairly simple.